### DISCUSSION TITLE
Invalid window geometry size error when taking focus off of the game

### DISCUSSION THREAD

**Discussion by The1Divider** [ppy/osu#17284] [Q&A] [answered]
Taking focus off of the game (`alt + tab`, `ctrl + esc`, `meta`, etc) results in a crash
```
xdg_surface@32: error -1: invalid window geometry size (-8x-39)
```
(Not sure if its dependant on resolution/other factors but the values are consistent between instant + method used to crash)
### Using:
Kubuntu 21.10 with Wayland
Intel i5-9300H w/t 8 GB memory
### Logs
[database.log](https://github.com/ppy/osu/files/8274168/database.log)
[network.log](https://github.com/ppy/osu/files/8274169/network.log)
[performance.log](https://github.com/ppy/osu/files/8274170/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8274171/runtime.log)

**Comment by bdach** [comment, marked as answer]
what compositor is this?

wayland is generally going to only be as good as SDL support of it is and that is still spotty. likely not much we can do in this repository about it, may need to report upstream.

**Reply by The1Divider** [reply]
Not sure where i'd find that information?

Ill try with a x11 environment in a minute, hopefully it is just a wayland issue

**Reply by bdach** [reply]
sorry, /s/compositor/window manager. the compositor is obviously wayland, just not sure which WM, as it may be WM-specific. command line tools like `screenfetch` / `neofetch` can show this.

**Reply by The1Divider** [reply]
Looks like it's KWin? Seems to work fine with X11

**Reply by bdach** [reply]
do you mind checking whether you have the newest version of kwin available for your distro? [this SDL PR thread](https://github.com/libsdl-org/SDL/pull/5064#issuecomment-987781687) suggests that there may be some relevant upstream fixes to kwin.

**Reply by The1Divider** [reply]
Ah apologies that did it, kubuntu had it at 5.22.5 - had to add the backlogs ppa but it seems to work now thanks.
